### PR TITLE
incorporate per-container resources and link to parent pod-spec into pod instance status

### PR DESCRIPTION
**Pull request by jdef**
/cc @orlandohohmeier

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/raml/PodStatusConversion.scala [hunk 1]
```diff
@@ -32,6 +32,16 @@ trait PodStatusConversion {
     // the Mesos task ID itself
     val displayName = task.taskId.containerName.getOrElse(task.taskId.mesosTaskId.getValue)
 
+    val resources: Option[Resources] = {
+      import InstanceStatus._
+      task.status.taskStatus match {
+        case Staging | Starting | Running | Reserved | Unreachable | Killing =>
+          maybeContainerName.flatMap(pod.container(_)).map(_.resources)
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/raml/PodStatusConversion.scala]: Use `pod.container(task).map(_.resources)`

File: src/main/scala/mesosphere/marathon/raml/PodStatusConversion.scala [hunk 2]
```diff
@@ -80,6 +93,7 @@ trait PodStatusConversion {
       networks = networkStatus,
       containers = containerStatus,
       message = message,
+      specReference = Some(s"/v2/pods${pod.id.canonicalPath()}::versions/${instance.runSpecVersion.toOffsetDateTime}"),
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/raml/PodStatusConversion.scala]: We should already have the canonicalPath here (no need to recompute).

File: docs/docs/rest-api/public/api/v2/types/podStatus.raml [hunk 3]
```diff
@@ -159,6 +163,10 @@ types:
       containers?:
         type: ContainerStatus[]
         description: status for each running container of this instance.
+      specReference?:
```
Comment (aquamatthias) [inline docs/docs/rest-api/public/api/v2/types/podStatus.raml]: Like the idea of pointing to the uri.
Do we have a runSpecVersion on the instance level?

Comment (jdef) [inline docs/docs/rest-api/public/api/v2/types/podStatus.raml]: we don't report the runSpecVersion as an instance-status field. we could. if we do, would you like that in this PR or a follow-up PR?

**Pull request merged**